tests/unit/test_helpers.py

```diff
@@ -227,10 +227,78 @@ def test_get_version_string(self):
             self.assertEqual(get_version_string(test[0]), test[1])
 
     def test_get_name(self):
-        '''Test name extraction from Dict via get_name helper method.'''
+        """Test name extraction from Dict via get_name helper method."""
         self.assertEqual(get_name({'family': 'Chrome'}), 'Chrome')
         self.assertEqual(get_name({'family': 'Mac OS X'}), 'Mac OS X')
         self.assertEqual(get_name({'family': 'Other'}), 'Unknown')
 
+    def test_form_type(self):
+        """Define which type of request for /issues/new."""
+        with webcompat.app.test_request_context(
+                '/issues/new?url=http://example.com/',
+                method='GET'):
+            self.assertEqual(form_type(flask.request), 'prefill')
+        with webcompat.app.test_request_context(
+                '/issues/new',
+                json={'url': 'http://example.com/'},
+                method='POST'):
+            self.assertEqual(form_type(flask.request), 'prefill')
+        with webcompat.app.test_request_context(
+                '/issues/new',
+                content_type='multipart/form-data',
+                data={'foo': 'blah'},
+                method='POST'):
+            self.assertEqual(form_type(flask.request), 'create')
+        with webcompat.app.test_request_context(
+                '/issues/new',
+                data={'url': 'http://example.com/'},
+                method='POST'):
+            self.assertEqual(form_type(flask.request), None)
+
+    def test_prepare_form_get(self):
+        """Extract information of a form request with a GET."""
+        form_data = {'extra_labels': [u'type-stylo'],
+                     'src': u'web',
+                     'user_agent': u'Burger',
+                     'url': u'http://example.net/',
+                     }
+        with webcompat.app.test_request_context(
+                '/issues/new?url=http://example.net/&src=web&label=type-stylo',
+                method='GET',
+                headers={'User-agent': 'Burger'}):
+            self.assertEqual(prepare_form(flask.request), form_data)
+        # Testing that we keep even when some parameters are not defined.
+        with webcompat.app.test_request_context(
+                '/issues/new?src=web&label=type-stylo',
+                method='GET',
+                headers={'User-agent': 'Burger'}):
+            # URL is not defined
+            form_data['url'] = None
```

Comment:
because this is the way this is done.
Explanation this is a GET. So this is triggering a call to the form which prefills some values. Here we have parameters for a couple of data, not including URL. This is very similar to a case where you just do a GET without parameters on the `/issues/new`. A URL is not defined. I'm just checking here that a missing param doesn't break anything.

We could change the business rule and says if there are parameters, we need at least this and this and this parameter together or we bail out. 

Note also there is a validation process later to check if the form is valid and url is a required parameter, aka once it has been programmatically entered or manually entered.

Reply:
> How does this pass with a missing `url`? I don't think I understand.

Big SIGH.

This is just me being slow 🐢. Of course no url will match no url.